node/src/reactor/joiner.rs

```diff
@@ -566,6 +570,10 @@ impl reactor::Reactor for Reactor {
             }
         };
 
+        let merkle_tree_hash_activation = chainspec_loader
+            .chainspec()
```

Comment:
```suggestion
        let merkle_tree_hash_activation = chainspec
```

(And probably in some more places in this function.)

Reply:
The `chainspec` binding is consumed by the `move` closure above and cannot be used any longer.

Introducing a new binding will not improve clarity that much, imho.